File: ios-native/PearBrowser/Sources/UI/Screens/TemplatePickerScreen.swift
//  PearBrowser — TemplatePickerScreen.swift
//
//  SwiftUI mirror of app/screens/TemplatePickerScreen.tsx.
//  Pure UI — no RPC. User picks one of 5 starter templates to
//  prefill the SiteEditorScreen with sensible default blocks.

import SwiftUI

struct SiteTemplate: Identifiable {
    let id: String
    let name: String
    let description: String
    let preview: String
    /// JSON-encodable block list passed to the editor.
    let blocks: [[String: Any]]
    let theme: [String: String]
}

struct TemplatePickerScreen: View {
    let onSelect: (SiteTemplate) -> Void
    let onBack: () -> Void

    var body: some View {
        VStack(spacing: 0) {
            ScreenHeader("Pick a template", onBack: onBack)

            ScrollView {
                VStack(spacing: 12) {
                    Text("Start with a template or pick Blank to build from scratch.")
                        .font(.system(size: 13))
                        .foregroundStyle(PearColors.textSecondary)
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(.bottom, 4)
                    ForEach(Self.templates) { tmpl in
                        TemplateCard(template: tmpl, onTap: { onSelect(tmpl) })
                    }
                }
                .padding(16)
            }
        }
        .background(PearColors.bg)
    }

    static let templates: [SiteTemplate] = [
        SiteTemplate(
            id: "blank",
            name: "Blank",
            description: "Start from scratch",
            preview: "{ }",
            blocks: [
                ["type": "heading", "text": "My Website", "level": 1],
                ["type": "text", "text": ""]
            ],
            theme: defaultTheme
        ),
        SiteTemplate(
            id: "personal",
            name: "Personal",
            description: "About me page with bio and links",
            preview: "@",
            blocks: [
                ["type": "heading", "text": "Your Name", "level": 1],
                ["type": "text", "text": "A short bio about yourself. What you do, what you care about."],
                ["type": "divider"],
                ["type": "heading", "text": "Links", "level": 2],
                ["type": "link", "text": "GitHub", "href": "https://github.com/you"],
                ["type": "link", "text": "Twitter", "href": "https://twitter.com/you"],
                ["type": "link", "text": "Email", "href": "mailto:you@example.com"],
            ],
            theme: themeOverride(primary: "#4dabf7")
        ),
        SiteTemplate(
            id: "blog",
            name: "Blog",
            description: "Blog post with title, date, and content",
            preview: "B",
            blocks: [
                ["type": "heading", "text": "Blog Post Title", "level": 1],
                ["type": "text", "text": "Published on April 2026"],
                ["type": "divider"],
                ["type": "text", "text": "Your blog post content goes here. Write about anything."],
                ["type": "quote", "text": "A meaningful quote that supports your argument."],
                ["type": "text", "text": "Wrap up your post with a conclusion."],
            ],
            theme: themeOverride(fontFamily: "Georgia, serif")
        ),
        SiteTemplate(
            id: "portfolio",
            name: "Portfolio",
            description: "Showcase your work with sections",
            preview: "P",
            blocks: [
                ["type": "heading", "text": "Your Name — Portfolio", "level": 1],
                ["type": "text", "text": "Designer / Developer / Creator"],
                ["type": "divider"],
                ["type": "heading", "text": "Project 1", "level": 2],
                ["type": "text", "text": "Description of your first project."],
                ["type": "link", "text": "View Project", "href": "https://example.com"],
                ["type": "divider"],
                ["type": "heading", "text": "Contact", "level": 2],
                ["type": "link", "text": "Email me", "href": "mailto:you@example.com"],
            ],
            theme: themeOverride(primary: "#4ade80")
        ),
        SiteTemplate(
            id: "landing",
            name: "Landing",
            description: "Simple product / idea landing page",
            preview: "L",
            blocks: [
                ["type": "heading", "text": "Your Product", "level": 1],
                ["type": "text", "text": "A one-line pitch that sells the idea."],
                ["type": "divider"],
                ["type": "heading", "text": "Why it matters", "level": 2],
                ["type": "text", "text": "Three sentences of context."],
                ["type": "link", "text": "Get Started", "href": "https://example.com"],
            ],
            theme: themeOverride(primary: "#facc15")
        ),
    ]

    private static let defaultTheme: [String: String] = [
        "primaryColor": "#ff9500",
        "backgroundColor": "#0a0a0a",
        "textColor": "#e0e0e0",
        "fontFamily": "-apple-system, sans-serif",
    ]

    private static func themeOverride(primary: String = "#ff9500",
                                      fontFamily: String = "-apple-system, sans-serif") -> [String: String] {
        [
            "primaryColor": primary,
            "backgroundColor": "#0a0a0a",
            "textColor": "#e0e0e0",
            "fontFamily": fontFamily,
        ]
    }
}

private struct TemplateCard: View {
    let template: SiteTemplate
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 14) {
                ZStack {
                    RoundedRectangle(cornerRadius: 12)
                        .fill(PearColors.surfaceElevated)
                        .frame(width: 56, height: 56)
                    Text(template.preview)
                        .font(.system(size: 24, weight: .bold, design: .monospaced))
                        .foregroundStyle(PearColors.accent)
                }
                VStack(alignment: .leading, spacing: 4) {
                    Text(template.name)
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundStyle(PearColors.textPrimary)
                    Text(template.description)
                        .font(.system(size: 12))
                        .foregroundStyle(PearColors.textSecondary)
                }
                Spacer()
                Text(">")
                    .font(.system(size: 18))
                    .foregroundStyle(PearColors.textMuted)
            }
            .padding(14)
            .background(PearColors.surface, in: RoundedRectangle(cornerRadius: 12))
        }
        .buttonStyle(.plain)
    }
}

File: android-native/app/src/main/java/com/pearbrowser/app/ui/screens/TemplatePickerScreen.kt
package com.pearbrowser.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.pearbrowser.app.ui.theme.PearColors

/**
 * TemplatePickerScreen — mirror of ios-native `TemplatePickerScreen.swift`.
 *
 * Pure UI — no RPC. The user picks one of 5 starter templates; the caller
 * creates the site (CMD_CREATE_SITE) and lands in SiteEditorScreen with the
 * template's blocks + theme prefilled.
 *
 * Template block/theme shapes mirror backend/site-manager.js `_renderBlocks`
 * and `_renderThemeCss` exactly.
 */
data class SiteTemplate(
    val id: String,
    val name: String,
    val description: String,
    val preview: String,
    val blocks: List<EditorBlock>,
    val theme: SiteTheme,
)

object SiteTemplates {
    private val defaultTheme = SiteTheme.default

    private fun themed(primary: String, fontFamily: String = "-apple-system, sans-serif") =
        SiteTheme(
            id = "template-$primary",
            label = "Template",
            primaryColor = primary,
            fontFamily = fontFamily,
        )

    val all: List<SiteTemplate> = listOf(
        SiteTemplate(
            id = "blank",
            name = "Blank",
            description = "Start from scratch",
            preview = "{ }",
            blocks = listOf(
                EditorBlock(type = BlockType.Heading, text = "My Website", level = 1),
                EditorBlock(type = BlockType.Text),
            ),
            theme = defaultTheme,
        ),
        SiteTemplate(
            id = "personal",
            name = "Personal",
            description = "About me page with bio and links",
            preview = "@",
            blocks = listOf(
                EditorBlock(type = BlockType.Heading, text = "Your Name", level = 1),
                EditorBlock(type = BlockType.Text, text = "A short bio about yourself. What you do, what you care about."),
                EditorBlock(type = BlockType.Divider),
                EditorBlock(type = BlockType.Heading, text = "Links", level = 2),
                EditorBlock(type = BlockType.Link, text = "GitHub", href = "https://github.com/you"),
                EditorBlock(type = BlockType.Link, text = "Twitter", href = "https://twitter.com/you"),
                EditorBlock(type = BlockType.Link, text = "Email", href = "mailto:you@example.com"),
            ),
            theme = themed("#4dabf7"),
        ),
        SiteTemplate(
            id = "blog",
            name = "Blog",
            description = "Blog post with title, date, and content",
            preview = "B",
            blocks = listOf(
                EditorBlock(type = BlockType.Heading, text = "Blog Post Title", level = 1),
                EditorBlock(type = BlockType.Text, text = "Published on April 2026"),
                EditorBlock(type = BlockType.Divider),
                EditorBlock(type = BlockType.Text, text = "Your blog post content goes here. Write about anything."),
                EditorBlock(type = BlockType.Quote, text = "A meaningful quote that supports your argument."),
                EditorBlock(type = BlockType.Text, text = "Wrap up your post with a conclusion."),
            ),
            theme = themed("#ff9500", fontFamily = "Georgia, serif"),
        ),
        SiteTemplate(
            id = "portfolio",
            name = "Portfolio",
            description = "Showcase your work with sections",
            preview = "P",
            blocks = listOf(
                EditorBlock(type = BlockType.Heading, text = "Your Name — Portfolio", level = 1),
                EditorBlock(type = BlockType.Text, text = "Designer / Developer / Creator"),
                EditorBlock(type = BlockType.Divider),
                EditorBlock(type = BlockType.Heading, text = "Project 1", level = 2),
                EditorBlock(type = BlockType.Text, text = "Description of your first project."),
                EditorBlock(type = BlockType.Link, text = "View Project", href = "https://example.com"),
                EditorBlock(type = BlockType.Divider),
                EditorBlock(type = BlockType.Heading, text = "Contact", level = 2),
                EditorBlock(type = BlockType.Link, text = "Email me", href = "mailto:you@example.com"),
            ),
            theme = themed("#4ade80"),
        ),
        SiteTemplate(
            id = "landing",
            name = "Landing",
            description = "Simple product / idea landing page",
            preview = "L",
            blocks = listOf(
                EditorBlock(type = BlockType.Heading, text = "Your Product", level = 1),
                EditorBlock(type = BlockType.Text, text = "A one-line pitch that sells the idea."),
                EditorBlock(type = BlockType.Divider),
                EditorBlock(type = BlockType.Heading, text = "Why it matters", level = 2),
                EditorBlock(type = BlockType.Text, text = "Three sentences of context."),
                EditorBlock(type = BlockType.Link, text = "Get Started", href = "https://example.com"),
            ),
            theme = themed("#facc15"),
        ),
    )

    fun byId(id: String): SiteTemplate = all.firstOrNull { it.id == id } ?: all.first()
}

@Composable
fun TemplatePickerScreen(
    onSelect: (SiteTemplate) -> Unit,
    onBack: () -> Unit,
) {
    Column(
        Modifier
            .fillMaxSize()
            .background(PearColors.Bg),
    ) {
        Row(
            Modifier
                .fillMaxWidth()
                .background(PearColors.Bg)
                .padding(horizontal = 10.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            TextButton(onClick = onBack) {
                Text("< Back", color = PearColors.Accent, fontSize = 14.sp)
            }
            Text(
                "Pick a template",
                color = PearColors.TextPrimary,
                fontSize = 22.sp,
                fontWeight = FontWeight.Bold,
            )
        }

        Column(
            Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
        ) {
            Text(
                "Start with a template or pick Blank to build from scratch.",
                color = PearColors.TextSecondary,
                fontSize = 13.sp,
            )
            Spacer(Modifier.height(12.dp))
            SiteTemplates.all.forEach { template ->
                TemplateCard(template = template, onClick = { onSelect(template) })
                Spacer(Modifier.height(12.dp))
            }
        }
    }
}

@Composable
private fun TemplateCard(template: SiteTemplate, onClick: () -> Unit) {
    Row(
        Modifier
            .fillMaxWidth()
            .background(PearColors.Surface, RoundedCornerShape(12.dp))
            .clickable(onClick = onClick)
            .padding(14.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center,
            modifier = Modifier
                .size(56.dp)
                .background(PearColors.SurfaceElevated, RoundedCornerShape(12.dp)),
        ) {
            Text(
                template.preview,
                color = PearColors.Accent,
                fontSize = 24.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = FontFamily.Monospace,
            )
        }
        Spacer(Modifier.width(14.dp))
        Column(Modifier.weight(1f)) {
            Text(
                template.name,
                color = PearColors.TextPrimary,
                fontSize = 16.sp,
                fontWeight = FontWeight.SemiBold,
            )
            Spacer(Modifier.height(4.dp))
            Text(template.description, color = PearColors.TextSecondary, fontSize = 12.sp)
        }
        Text(">", color = PearColors.TextMuted, fontSize = 18.sp)
    }
}
